TFlippy_TerritoryControl_Misc_Dev/Entities/Equipment/ParachutePack/parachutepack_effect.as

```diff
@@ -0,0 +1,147 @@
+#include "RunnerTextures.as"
+//Script by Gingerbeard
+
+void onInit(CBlob@ this)
+{
+	this.addCommandID("deploy_chute");
+
+	if(this.get_string("reload_script") != "parachutepack")
+	{
+		UpdateScript(this);
+	}
+}
+
+void UpdateScript(CBlob@ this)
+{
+	CSpriteLayer@ pack = this.getSprite().addSpriteLayer("pack", "parachutepack.png", 16, 16);
+	if (pack !is null)
+	{
+		pack.SetVisible(true);
+		pack.SetRelativeZ(-2);
+		pack.SetOffset(Vec2f(0, 0));
+
+		if(this.getSprite().isFacingLeft())
+		{
+			pack.SetFacingLeft(true);
+		}
+	}
+
+	int teamnum = this.getTeamNum();
+	CSpriteLayer@ parachute = this.getSprite().addSpriteLayer("parachute", "parachutepack", 32, 32, teamnum, 0);
+	if (parachute !is null)
+	{
+		Animation@ anim = parachute.addAnimation("default", 0, true);
+		anim.AddFrame(1);
+		parachute.SetOffset(Vec2f(0.0f, - 17.0f));
+		parachute.SetVisible(false);
+	}
+}
+
+void GetButtonsFor(CBlob@ this, CBlob@ caller)
+{
+	CBitStream params;
+	CBlob@ carried = caller.getCarriedBlob();
+	this.addCommandID("deploy_chute");
+
+	bool candeploy = (this.isOnGround() || this.isInWater() || this.isAttached() || this.isOnLadder());
+	bool deployavailable = getGameTime() >= this.get_u32("next use");
+
+	if (caller is this && !candeploy && deployavailable && !this.hasTag("parachute"))
+	{
+		caller.CreateGenericButton(6, Vec2f(0.0f, 0.0f), this, this.getCommandID("deploy_chute"), "Deploy!", params);
+	}
+}
+
+void onCommand(CBlob@ this, u8 cmd, CBitStream @params)
+{
+	if (cmd == this.getCommandID("deploy_chute"))
+	{
+		this.Tag("parachute");
+	}
+}
+
+void onTick(CBlob@ this)
+{
+	bool candeploy = (this.isOnGround() || this.isInWater() || this.isAttached() || this.isOnLadder());
+	bool deployavailable = getGameTime() >= this.get_u32("next use");
+
+	if(this.get_string("reload_script") == "parachutepack")
+	{
+		UpdateScript(this);
+		this.set_string("reload_script", "");
+	}
+
+	CSpriteLayer@ pack = this.getSprite().getSpriteLayer("pack");
+	if (pack !is null)
```

Comment:
move isClient check to here (Server does not need to run any CSprite code)

Reply:
isClient has been moved.